pkg/clients/postgres/clients.go

```diff
@@ -537,20 +510,26 @@ func ConstructSearchQuery(pm clients.Page) (string, string) {
 	if pm.Id != "" {
 		query = append(query, "id ILIKE '%' || :id || '%'")
 	}
-
-	if len(query) > 0 {
-		emq = fmt.Sprintf("WHERE %s", strings.Join(query, " AND "))
+	if pm.Tag != "" {
+		query = append(query, "EXISTS (SELECT 1 FROM unnest(tags) AS tag WHERE tag ILIKE '%' || :tag || '%')")
+	}
+	if len(query) == 0 && len(pm.IDs) != 0 {
```

Comment:
We are treating IDs differently, i.e. taking it into account only if other criteria is not there, making `buildQueryConditions` mask the logic of processing IDs. Why?

Reply:
This is allows `searchClients` to retrieve basic info by id or search. So if either `name`, `tag`, `identify`, or `id` is set we don't retrieve by `id`

Reply:
Why don't we still combine? If you want only IDs, you send only IDs in the query.

Reply:
We can but to get you properly, you are suggesting we combine both `searchClients` and `retrieveAllByIds` right?

Reply:
Eventually, that would be the outcome here, yes. I.e. Search users from this pool of IDs with this criteria. It's unlikely to need such a thing, but it makes `buildQueryConditions` consistent.